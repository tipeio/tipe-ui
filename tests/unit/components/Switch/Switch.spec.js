import { shallowMount } from '@vue/test-utils'
import TipeSwitch from '@/components/Switch'

describe('Progress.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        text: 'Share Location'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
  it('should have the correct label', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        text: 'Share Location'
      }
    })
    expect(wrapper.find('.label').text()).toBe('Share Location')
  })
  it('should have the correct sublabel', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        text: 'Share Location',
        sublabel: 'Share location while using the app'
      }
    })
    expect(wrapper.find('.sublabel').text()).toBe(
      'Share location while using the app'
    )
  })
  it('should not render with a sublabel', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        text: 'Share Location'
      }
    })
    expect(wrapper.find('.sublabel').exists()).toBe(false)
  })
  it('should have the correct value prop', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        text: 'Share Location',
        value: true
      }
    })
    expect(wrapper.props().value).toBe(true)
  })
})
