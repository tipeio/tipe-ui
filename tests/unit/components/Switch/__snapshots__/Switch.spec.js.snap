// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch <template> renders 1`] = `
<div data-tipe-ui="TipeSwitch" class="wrapper">
  <div class="row"><label class="switch medium"><input id="Share Location-switch" name="location" tabindex="0" type="checkbox"> <span class="slider round medium"></span></label> <label for="Share Location-switch" class="label">
      Share Location
    </label></div>
</div>
`;
